### ISSUE TITLE
DoubleUpDown Dependency Property Already Registered

### ISSUE THREAD

**Issue by xceedsoftware** [xceedsoftware/wpftoolkit#1017]
<b>cellomanpierce[CodePlex]</b> <br />I am getting the following Xaml designer error when I use two different types of UpDown controls in the same document (In my case I have several IntegerUpDown controls and a DoubleUpDown). It seems that the dependency property is being registered multiple
 times, though it appears to be set up correctly as a static property in CommonNumericUpDownltTgt. If I get rid of the DoubleUpDown control, the error goes away, but if I leave the DoubleUpDown and get rid of the IntegerUpDown controls instead, I get a
 similar error, but referencing the 'AutoMoveFocus' property (contained in UpDownBaseltTgt). This error doesn't prevent me from compiling and running my project just fine, but it does throw the error in the designer and prevent the design display from displaying
 my controls.

Error

Severity Code Description Project File Line Suppression State 'ParsingNumberStyle' property was already registered by 'Xceed_Wpf_Toolkit_CommonNumericUpDown`1_39_157692670'.

**Comment by xceedsoftware** [comment]
<b>reasra[CodePlex]</b> <br />Can confirm shui's post:
Problem occurs with 2.9 and 3.0, only when x64 is the target.
On all xaml that uses at least two controls that inherit UpDownBase.

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

Yes I can reproduce this with the platform set to x64 under VS2015.
I suspect this is a Visual Studio XAML parser problem.
VS2010 doesn't have this problem for the same xaml code. 

――――
Get more controls, features, updates and technical support with 
Xceed Toolkit Plus for WPF

**Comment by xceedsoftware** [comment]
<b>shui[CodePlex]</b> <br />I have reproduced this bug with VS 2015. 
Both Xceed.Wpf.Toolkit.dll v2.9 and Xceed.Wpf.Toolkit.dll v3.0 have this bug. 

To reproduce the bug, should use x64 as the platform, dont use x86 or AnyCPU!

**Comment by xceedsoftware** [comment]
<b>tibertiger[CodePlex]</b> <br />Hello, 

Any updates ? I hit this very issue using VS 2015 Enterprise, x64 while using the designer. The error does not occur while running the application.

**Comment by xceedsoftware** [comment]
<b>thecake[CodePlex]</b> <br />I tried to create a test project, but that did not throw the exception. 

It looks like this is somehow related to the VS designer not working properly when projects get too big?

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

Can you submit a test project ?
I cannot reproduce this error in the Designer of VS 2015 Enterprise Update 3 with Xceed.Wpf.Toolkit.dll v2.9 Community Edition.


I use the Toolbox to add 2 IntegerUpDowns and 1 DoubleUpDown. No errors are displayed and the project compiles and executes as expected.

**Comment by xceedsoftware** [comment]
<b>thecake[CodePlex]</b> <br />I have the same error with VS Enterprise 2015 Update 3.
There are multiple IntegerUpDowns but the error only appears with the DoubleUpDown control.

**Comment by xceedsoftware** [comment]
<b>Shujee[CodePlex]</b> <br />I can reproduce it too. I'm using VS2015 Community. I have one IntegerUpDown two DateTimeUpDown controls.

**Comment by Bo98** [comment]
I believe this may be from the use of generics. A generic-less baseclass that contains the dependency properties may prevent this issue.

It is odd that it doesn't happen with "Any CPU", but it happens in both VS2015 and VS2017.

**Comment by wonkoRT** [comment]
Can reproduce this as well with VS2015Pro/U3. Tested with x64 build and IntegerUpDown and DoubleUpDown. I found a post which might be related [here](https://stackoverflow.com/questions/3629403/set-dependency-properties-in-xaml-when-the-base-class-is-generic)

**Comment by XceedBoucherS** [comment]
Hi,
The idea from the link is good, but not easy to implement in current Toolkit code : it contains multi-generic base classes.

This is a Visual studio XAML parser error.

**Comment by bh4017** [comment]
I get a similar complaint in VS2017 Pro.  Only when x64 is the target.

> 'ParsingNumberStyle' property was already registered by 'Xceed_Wpf_Toolkit_CommonNumericUpDown`1_9_144497016'.

**Comment by amputek** [comment]
Getting this in VS2017 (15.8.7), only with a x64 target.

> 'IsInvalid' property was already registered by Xceed_Wpf_Toolkit_CommonNumericUpDown`1_3_108099632'.

**Comment by amputek** [comment]
+1. Getting this in VS2017 with a x64 target.

> 'IsInvalid' property was already registered by 'Xceed_Wpf_Toolkit_CommonNumericUpDown`1_3_108099632'.

**Comment by johmarjac** [comment]
Still getting this, wow, why is this still not fixed after years?

**Comment by Dirkster99** [comment]
...because their workflow (picking up an issue in open source version and fixing it 2-4 releases later) **does not support** [complex resolution/retest scenarious](https://github.com/Dirkster99/AvalonDock/issues/27) where feedback from reporters is required to ensure a valid resolution...

**Comment by hcorg** [comment]
Still present in VS2019....

**Comment by TheWhiteAmbit** [comment]
Yes, I can confirm this is still a thing with 3.6 and VS2017 and AnyCPU with x64 preferences. Seems to occure whenever IntegerUpDown and DoubleUpDown are mixed in a UserControl.

**Comment by BitmineBarry** [comment]
I am seeing the issue in VS2019 with just the DoubleUpDown control mixed with standard xaml controls. i.e. No IntegerUpDown in the XAML

**Comment by BitmineBarry** [comment]
This is still not easy to reproduce. 
Attempting to make a simple project to confirm the issue I have mixed IntegerUpDown and DoubleUpDown controls into a small project I use for GUI experimentation and Im not seeing an issue. 
If I manage to isolate the cause in my main project I will post the info here.

**Comment by cwenger** [comment]
@BitmineBarry Make sure you are using the 'x64' build configuration. With 'Any CPU' it seems to work fine. I just tried and was able to reproduce the issue very quickly with a new WPF project.

**Comment by qqblack** [comment]
I can reproduce this problem in my project only in x64.But,I can't reproduce it in a new project even in x64 platform.

My VS version is VS2019 Enterprise. And I have tested Xceed_Wpf_Toolkit 3.8.1 and  4.0.1

**Comment by TheWhiteAmbit** [comment]
The problem still persists in release 4.0.1 and VS2017. Only in x64 builds. This is a really nasty bug making use of WpfToolkit impossible where a visual UI Editor is needed - so almost everywhere. It is also no solution to switch to x86 for editing, since the bug remains once it came up until you close VS and reload. So once I need to run in x64 the bug is there. It also seems to slow down VS after a while to a point where it becomes unusable. Please really fix this bug or WpfToolkit will be marked "don't ever use in projects again" in my heads lists of bad software. It has been three years now and became a constant headache in my project. If it is not fixed within a few months, I will have to replace all WpfToolkit with something similar. This is work I would love not to do!

**Comment by wonkoRT** [comment]
I can still reproduce this in a new project created with VS2019 V16.7.2 and targeting x64 platform. Nothing changed.
I agree with [XceedBoucherS](https://github.com/xceedsoftware/wpftoolkit/issues/1017#issuecomment-324101097) that this *could* be [fixed](https://developercommunity.visualstudio.com/search.html?f=&type=problem&type=problem&c=&redirect=search%2Fsearch&sort=relevance&q=xaml+designer) in [VS](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio) but apparently it's not too high on MS prio list. [Replacing some of the genericness and get less dry](https://stackoverflow.com/questions/3629403/set-dependency-properties-in-xaml-when-the-base-class-is-generic) could work around it.
